nano/node/websocket.cpp

```diff
@@ -4,6 +4,54 @@
 #include <nano/node/node.hpp>
 #include <nano/node/websocket.hpp>
 
+nano::websocket::confirmation_options::confirmation_options (boost::property_tree::ptree const & options_a, nano::node & node_a) :
+all_local_accounts (options_a.get<bool> ("all_local_accounts", false)),
+node (node_a)
+{
+	auto accounts_l (options_a.get_child_optional ("accounts"));
+	if (accounts_l)
+	{
+		for (auto account_l : *accounts_l)
+		{
+			// Check if the account is valid, but no error handling if it's not, simply not added to the filter
+			nano::account result_l (0);
+			if (!result_l.decode_account (account_l.second.data ()))
+			{
+				accounts.insert (account_l.second.data ());
+			}
+		}
+	}
+}
+
+bool nano::websocket::confirmation_options::should_filter (nano::websocket::message const & message_a) const
+{
+	bool should_filter_l (true);
+	auto source_text_l (message_a.contents.get<std::string> ("message.account"));
+	auto destination_opt_l (message_a.contents.get_optional<std::string> ("message.block.link_as_account"));
+	if (!destination_opt_l)
+	{
+		destination_opt_l = message_a.contents.get_optional<std::string> ("message.block.destination");
+	}
+	assert (destination_opt_l);
+	auto destination_text (destination_opt_l.get ());
+	if (all_local_accounts)
+	{
+		auto transaction_l (node.wallets.tx_begin_read ());
+		nano::account source_l (0), destination_l (0);
+		assert (!source_l.decode_account (source_text_l));
```

Comment:
I guess this line & one below can work only in debug builds

Reply:
So, if functions inside of assert () are ignored in Release build, then both source_l & destination_l remain 0 I guess

Reply:
I'm switching to release_assert instead, or do you rather not have it?

Reply:
auto error (...)
assert (!error)

Reply:
Thanks, eheh